Comp2023/src/main/java/frc/robot/commands/DriveLimelightPath.java

```diff
@@ -86,35 +97,58 @@ public int getSignFromId(int targetId)
     return (targetId <= 4) ? -1 : 1;
   }
 
-  public Pose2d calculateTarget(int targetId, VIConsts.VIGoalDirection goalDirection)
+  public Pose2d getGoalPose(VIConsts.VIGoalDirection goalDirection)
   {
+    int targetId = m_vision.getTargetID( );
+
+    if ((!m_vision.isAprilTagValid(targetId)) || (targetId < 0))
+    {
+      return null;
+    }
+
     Pose2d targetPose = VIConsts.kAprilTagPoses.get(targetId);
     double goalXValue = 0;
     double goalYValue = 0;
     String strName;
 
-    goalXValue = targetPose.getX( ) + getSignFromId(targetId) * VIConsts.kAdjustPathX;
+    if (targetId == 4 || targetId == 5)
+    {
+      goalXValue = targetPose.getY( ) + getSignFromId(targetId) * (VIConsts.kAdjustSubPathX);
+    }
+    else
+    {
+      goalXValue = targetPose.getX( ) + getSignFromId(targetId) * (VIConsts.kAdjustPathX);
+    }
 
     switch (goalDirection)
     {
       default :
       case DIRECTION_LEFT :
         strName = "LEFT";
-        goalYValue = targetPose.getY( ) + getSignFromId(targetId) * VIConsts.kAdjustPathY;
+        goalYValue = targetPose.getY( ) + getSignFromId(targetId) * (VIConsts.kAdjustPathY);
+        if (targetId == 4 || targetId == 5)

```

Comment:
Something to dig into later, but it's not quite clear why we'd get goalYValue from the target pose and then do it again for 4 and 5. An if then else would have been clearer.

Reply:
Or even just hiding this in a method that was well commented.